runelite-client/src/main/java/net/runelite/client/util/HotkeyListener.java

```diff
@@ -42,15 +42,28 @@
 	@Override
 	public void keyTyped(KeyEvent e)
 	{
+		if (Keybind.getModifierForKeyCode(e.getKeyCode()) == null && isPressed)
+		{
+			e.consume();
+		}
 	}
 
 	@Override
 	public void keyPressed(KeyEvent e)
 	{
 		if (keybind.get().matches(e))
 		{
+			boolean wasPressed = isPressed;
 			isPressed = true;
-			hotkeyPressed();
+			if (wasPressed)
```

Comment:
It should be `if (!wasPressed)`, or else it still repeats when the key is held down and even skips the first press.